game/lib/input.py

```diff
@@ -0,0 +1,137 @@
+                 text_color=DEFAULT_TEXT_COLOR, input_border=DEFAULT_INPUT_BORDER, border_color=DEFAULT_BORDER_COLOR,
+                 focus_color=DEFAULT_FOCUS_COLOR, nofocus_color=DEFAULT_NOFOCUS_COLOR, is_password=False):
+        if pos_x < 1:
+            suf = pygame.display.get_surface()
+            if suf is not None:
+                self.__pos_x = pos_x * suf.get_width()
+            else:
+                self.__pos_x = DEFAULT_POS_X
+        else:
+            self.__pos_x = pos_x
+        if pos_y < 1:
+            suf = pygame.display.get_surface()
+            if suf is not None:
+                self.__pos_y = pos_y * suf.get_height()
+            else:
+                self.__pos_y = DEFAULT_POS_Y
+        else:
+            self.__pos_y = pos_y
+        if width < 1:
+            suf = pygame.display.get_surface()
+            if suf is not None:
+                self.__width = width * suf.get_width()
+            else:
+                self.__width = DEFAULT_WIDTH
+        else:
+            self.__width = width
+        if height < 1:
+            suf = pygame.display.get_surface()
+            if suf is not None:
+                self.__height = height * suf.get_height()
+            else:
+                self.__height = DEFAULT_HEIGHT
+        else:
+            self.__height = height
+        self.__is_password = is_password
+        self.__label = label
+        self.__value = ''
+        self.__text_size = text_size
+        self.__label_text_size = label_size
+        self.__font_label = pygame.font.SysFont(FONT_STYLE, self.__label_text_size)
+        self.__label_color = label_color
+        self.__text_color = text_color
+        self.__active = False
+        self.__font_text = pygame.font.SysFont(FONT_STYLE, self.__text_size)
+        self.__text_start = 0
+        self.__input_border = input_border
+        self.__border_color = border_color
+        self.__focus_color = focus_color
+        self.__nofocus_color = nofocus_color
+
+    def check_mouse(self, input_rect):
+        if pygame.mouse.get_pressed()[0] and input_rect.collidepoint(pygame.mouse.get_pos()):
+            self.__active = True
+        elif pygame.mouse.get_pressed()[0] and not input_rect.collidepoint(pygame.mouse.get_pos()):
+            self.__active = False
+
+    def check_user_input(self, events):
+        for event in events:
+            if event.type == pygame.KEYDOWN:
+                if event.key == pygame.K_BACKSPACE and len(self.__value) > 0:
+                    self.__value = self.__value[:-1]
+                    if self.__text_start > 0:
+                        self.__text_start = self.__text_start - 1
+                elif event.unicode.lower() in KEYS:
+                    self.__value = self.__value + event.unicode
+
+    def get_value(self):
+        return self.__value
+
+    def get_text(self):
+        if self.__is_password:
+            str = ''
+            for i in range(self.__text_start, len(self.__value)):
```

Comment:
97-99 można zapisać:
str = "".join(['*' for i in range(self.__text_start, len(self.__value))])